azurelinuxagent/common/utils/archive.py

```diff
@@ -58,13 +58,15 @@
 #   2018-04-06T08:21:37.142697.zip
 #   2018-04-06T08:21:37.142697_incarnation_N
 #   2018-04-06T08:21:37.142697_incarnation_N.zip
+#   2018-04-06T08:21:37.142697_N-M
+#   2018-04-06T08:21:37.142697_N-M.zip
 #
 # Current names
 #
-#   2018-04-06T08:21:37.142697_N-M
-#   2018-04-06T08:21:37.142697_N-M.zip
+#   2018-04-06T08-21-37__N-M
+#   2018-04-06T08-21-37__N-M.zip
 #
-_ARCHIVE_BASE_PATTERN = r"\d{4}\-\d{2}\-\d{2}T\d{2}:\d{2}:\d{2}\.\d+((_incarnation)?_(\d+|status)(-\d+)?)?"
+_ARCHIVE_BASE_PATTERN = r"\d{4}\-\d{2}\-\d{2}T\d{2}[:-]\d{2}[:-]\d{2}(\.\d+)?((_incarnation)?_+(\d+|status)(-\d+)?)?"
```

Comment:
we use '-' instead of ':' in the time, so why do we need` :` in regex pattern

Reply:
we need to match both the current and legacy formats